File: WPF_GiamDinhBaoHiemYTe/Repos/Model/Patietn_XML_data/XML2.cs
using CommunityToolkit.Mvvm.ComponentModel;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WPF_GiamDinhBaoHiem.Repos.Model
{
    public partial class XML2 : ObservableObject
    {
        [ObservableProperty] private int id;
        [ObservableProperty] private string? ma_Lk;
        [ObservableProperty] private int? stt;
        [ObservableProperty] private string? ma_Thuoc;
        [ObservableProperty] private string? ma_Pp_CheBien;
        [ObservableProperty] private string? ma_Cskcb_Thuoc;
        [ObservableProperty] private int? ma_Nhom;
        [ObservableProperty] private string? ten_Thuoc;
        [ObservableProperty] private string? don_Vi_Tinh;
        [ObservableProperty] private string? ham_Luong;
        [ObservableProperty] private string? duong_Dung;
        [ObservableProperty] private string? dang_Bao_Che;
        [ObservableProperty] private string? lieu_Dung;
        [ObservableProperty] private string? cach_Dung;
        [ObservableProperty] private string? so_Dang_Ky;
        [ObservableProperty] private string? tt_Thau;
        [ObservableProperty] private int? pham_Vi;
        [ObservableProperty] private decimal? tyle_Tt_Bh;
        [ObservableProperty] private decimal? so_Luong;
        [ObservableProperty] private decimal? don_Gia;
        [ObservableProperty] private decimal? thanh_Tien_Bv;
        [ObservableProperty] private decimal? thanh_Tien_Bh;
        [ObservableProperty] private decimal? t_NguonKhac_Nsnn;
        [ObservableProperty] private decimal? t_NguonKhac_Vtnn;
        [ObservableProperty] private decimal? t_NguonKhac_Vttn;
        [ObservableProperty] private decimal? t_NguonKhac_Cl;
        [ObservableProperty] private decimal? t_NguonKhac;
        [ObservableProperty] private decimal? muc_Huong;
        [ObservableProperty] private decimal? t_Bntt;
        [ObservableProperty] private decimal? t_Bncct;
        [ObservableProperty] private decimal? t_Bhtt;
        [ObservableProperty] private string? ma_Khoa;
        [ObservableProperty] private string? ma_Bac_Si;
        [ObservableProperty] private string? ma_Dich_Vu;
        [ObservableProperty] private string? ngay_Yl;
        [ObservableProperty] private int? ma_Pttt;
        [ObservableProperty] private int? nguon_Ctra;
        [ObservableProperty] private int? vet_Thuong_Tp;
        [ObservableProperty] private string? du_Phong;
        [ObservableProperty] private string? ngay_Th_Yl;
        [ObservableProperty] private bool isError;
    }
}

File: Backend/src/Repos/Models/XML_Model/XML2.js
const mongoose = require('mongoose');
const { convertDecimal128 } = require('../../../Utils/ConvertDecimal');


const XML2Schema = new mongoose.Schema({
  id:                   { type: Number, required: false },
  Ma_Lk:                { type: String, default: null },
  Stt:                  { type: Number, default: null },
  Ma_Thuoc:             { type: String, default: null },
  Ma_Pp_CheBien:        { type: String, default: null },
  Ma_Cskcb_Thuoc:       { type: String, default: null },
  Ma_Nhom:              { type: Number, default: null },
  Ten_Thuoc:            { type: String, default: null },
  Don_Vi_Tinh:          { type: String, default: null },
  Ham_Luong:            { type: String, default: null },
  Duong_Dung:           { type: String, default: null },
  Dang_Bao_Che:         { type: String, default: null },
  Lieu_Dung:            { type: String, default: null },
  Cach_Dung:            { type: String, default: null },
  So_Dang_Ky:           { type: String, default: null },
  Tt_Thau:              { type: String, default: null },
  Pham_Vi:              { type: Number, default: null },
  Tyle_Tt_Bh:           { type: mongoose.Schema.Types.Decimal128, default: null },
  So_Luong:             { type: mongoose.Schema.Types.Decimal128, default: null },
  Don_Gia:              { type: mongoose.Schema.Types.Decimal128, default: null },
  Thanh_Tien_Bv:        { type: mongoose.Schema.Types.Decimal128, default: null },
  Thanh_Tien_Bh:        { type: mongoose.Schema.Types.Decimal128, default: null },
  T_NguonKhac_Nsnn:     { type: mongoose.Schema.Types.Decimal128, default: null },
  T_NguonKhac_Vtnn:     { type: mongoose.Schema.Types.Decimal128, default: null },
  T_NguonKhac_Vttn:     { type: mongoose.Schema.Types.Decimal128, default: null },
  T_NguonKhac_Cl:       { type: mongoose.Schema.Types.Decimal128, default: null },
  T_NguonKhac:          { type: mongoose.Schema.Types.Decimal128, default: null },
  Muc_Huong:            { type: mongoose.Schema.Types.Decimal128, default: null },
  T_Bntt:               { type: mongoose.Schema.Types.Decimal128, default: null },
  T_Bncct:              { type: mongoose.Schema.Types.Decimal128, default: null },
  T_Bhtt:               { type: mongoose.Schema.Types.Decimal128, default: null },
  Ma_Khoa:              { type: String, default: null },
  Ma_Bac_Si:            { type: String, default: null },
  Ma_Dich_Vu:           { type: String, default: null },
  Ngay_Yl:              { type: String, default: null },
  Ma_Pttt:              { type: Number, default: null },
  Nguon_Ctra:           { type: Number, default: null },
  Vet_Thuong_Tp:        { type: Number, default: null },
  Du_Phong:             { type: String, default: null },
  Ngay_Th_Yl:           { type: String, default: null },
  chucdanh_id:          { type: Number, default: null }
}, {
  timestamps: true,
  strict: false
});

XML2Schema.set('toJSON',{
  transform: (_, ret) => convertDecimal128(ret),
});

XML2Schema.set('toObject', {
  transform: (_, ret) => convertDecimal128(ret),
});

module.exports = mongoose.model('XML2', XML2Schema);
